newt/resolve/resolve.go

```diff
@@ -644,6 +653,18 @@ func ResolveFull(
 
 	res.MasterSet.Rpkgs = r.rpkgSlice()
 
+	// We have now resolved all packages so go through them and emit warning
+	// when using link packages
+	for _, rpkg := range res.MasterSet.Rpkgs {
+		if rpkg.Lpkg.Type() != pkg.PACKAGE_TYPE_TRANSIENT {
+			continue
+		}
+
+		log.Warnf("Transient package %s used, update configuration "+
+			"to use target package instead (%s)",
+			rpkg.Lpkg.FullName(), rpkg.Lpkg.LinkedName())
```

Comment:
This warning might be confusing since "target" already means something else.  Maybe call it the "linked package" instead of "target package"?